pippy/hf/_SaveModule.py

```diff
@@ -141,7 +143,33 @@ def _save_params(submod: torch.nn.Module, checkpoint_dir: str) -> None:
     )
 
 
-def save_checkpoint(stage: Pipe, checkpoint_dir: str) -> None:
+def _save_optim_state(
+    optimizer: torch.optim.Optimizer, checkpoint_dir: str
+) -> None:
+    """
+    saves `optimizer`'s state_dict to disk.
+
+    Args:
+        optimizer(`torch.optim.Optimizer`): pytorch optimizer
+        checkpoint_dir(`str`): where to keep the checkpoint binaries
+    """
+    if not os.path.exists(checkpoint_dir):
+        os.mkdir(checkpoint_dir)
+    filepath = os.path.join(
+        checkpoint_dir, _get_binary_filename(dist.get_rank(), is_optim=True)
+    )
+    torch.save(
```

Comment:
You can directly just torch.save(optimizer.state_dict()).
The optimizer structure is different from model_state_dict if you print it out.

Reply:
What's the motivation of saving parameters individually? If there is a valid reason to do so, should we do the same thing for the optimizer state_dict, saving each optimizer state in a separate file? We should consider how users want the states to be saved and it is best to be consistent for model and optimizer.